game/missiongenerator/convoygenerator.py

```diff
@@ -32,17 +33,46 @@ def generate(self) -> None:
                 self.generate_convoy(convoy)
 
     def generate_convoy(self, convoy: Convoy) -> VehicleGroup:
-        group = self._create_mixed_unit_group(
-            convoy.name,
-            convoy.route_start,
-            convoy.units,
-            convoy.player_owned,
-        )
-        group.add_waypoint(
-            convoy.route_end,
-            speed=kph(40).kph,
-            move_formation=PointAction.OnRoad,
-        )
+        if self.game.settings.perf_moving_convoys:
+            group = self._create_mixed_unit_group(
+                convoy.name,
+                convoy.route_start,
+                convoy.units,
+                convoy.player_owned,
+            )
+
+            group.add_waypoint(
+                convoy.route_end,
+                speed=kph(40).kph,
+                move_formation=PointAction.OnRoad,
+            )
+
+        else:
+            # perf_moving_convoys is disabled, so add the convoy between the route start and route end.
+            # This option aims to remove long routes for ground vehicles between control points,
+            # since the CPU load for long routes on DCS is pretty heavy.
+            frontline = FrontLine(convoy.origin, convoy.destination)
+
+            # Select a segment roughly a third of the way from the origin towards the destination
+            # so the convoy spawns between the control points but is still close enough to the
+            # origin CP to be targeted by BAI flights and within the protection umbrella of the CP.
+            # Convoy start and end waypoints are not the same, so it'll move a little
+            # before stopping and hopefully find a road to drive on.
+            convoy_segment = int(0.3 * len(frontline.segments))
+
+            group = self._create_mixed_unit_group(
+                convoy.name,
+                frontline.segments[convoy_segment].point_a,
+                convoy.units,
+                convoy.player_owned,
+            )
```

Comment:
This could be nowhere remotely near the BAI target area. I don't think it's worth fighting all the edge cases in this when we could instead just spawn at the origin point. The convoys just barely leave their origin in the span of a mission anyway, so it's a more accurate representation to have them there than 1/3 of the way between bases anyway.

I don't think the code here needs to be any more complicated than an `if` around the existing `add_waypoint`.

Reply:
Implemented the suggested change.